### DISCUSSION TITLE
Video 20 Matrix Multiplication - Why did matmul() work for (1*3)* (1*3) matrix at 2:24:20?

### DISCUSSION THREAD

**Discussion by jsunecha** [mrdbourke/pytorch-deep-learning#773] [Q&A] [answered]
Hi everyone,

Thank you so much Daniel for this amazing course. I had a question- from my understanding, for matrix multiplication, the inner dimensions of the matrices being multiplied should match. However, at 2:24:20, you have a tensor
```python
tensor = torch.tensor([1, 2, 3])
```
that you have multiplied with itself. How did this still give an output (14), and not an error?

**Comment by sametucuncu** [comment, marked as answer]
If both tensors are 1-dimensional, the dot product (scalar) is returned. (resource: torch.matmul documentation)

**Reply by jsunecha** [reply]
Thank you @sametucuncu!
